### PR TITLE
support for local tar input, message .trigger extension, append log to trailer file, crds_context pinning

### PR DESCRIPTION
**Pull request by alphasentaurii**
- support for local tar inputs
- append process.txt to trailer
- add .trigger ext to proc message
- optional crds context pinning in assign_bestrefs (process.py)
- set crds context explicitly in Github Actions for testing

### CODE REVIEW

File: caldp/file_ops.py [hunk 1]
```diff
@@ -9,6 +9,44 @@
...
+    """
+    try:
+        tra1 = list(glob.glob(f"{output_path}/{ipppssoot.lower()}.tra"))
+        tra2 = list(glob.glob(f"{output_path}/{ipppssoot.lower()[0:5]}*.tra"))
+        if os.path.exists(tra1[0]):
+            trailer = tra1[0]
+        elif os.path.exists(tra2[0]):
+            trailer = tra2[0]
+        else:
+            log.error("Trailer file not found - skipping.")
```
Comment (bhayden53) [inline caldp/file_ops.py:37]: can you make the missing trailer an info instead of error?

Comment (alphasentaurii) [reply caldp/file_ops.py:37]: yes

Comment (bhayden53) [reply caldp/file_ops.py:37]: And I think if it's truly missing, what we want to do is create it, though we would want to tag up with Violet on that

File: caldp/process.py [hunk 2]
```diff
@@ -487,6 +504,8 @@ def process(self, files):
         unassoc = self.unassoc_files(files)
         if unassoc:
             self.run(self.stage1, *unassoc)
+            # if self.stage2:
```
Comment (bhayden53) [inline caldp/process.py:507]: if you leave this commented out doesn't it break auto-merge of Rob's PR?

Comment (bhayden53) [reply caldp/process.py:507]: I mean, if you uncomment it then I guess that **_IS_** Rob's PR

**Comment by bhayden53** [comment]
I would uncomment the stage2 code and singletons tests and we'll merge that here, then the other PR is deprecated and we can close without merge.

**Pull request merged**